src/widgetastic_patternfly4/clipboardcopy.py

```diff
@@ -0,0 +1,45 @@
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.widget import TextInput
+from widgetastic.widget import View
+
+from widgetastic_patternfly4.button import Button
+
+
+class BaseClipboardCopy:
+    BUTTON_LOC = ".//button[contains(@class, 'pf-c-button pf-m-control')]"
+    TEXT_LOC = ".//input[contains(@class, 'pf-c-form-control')]"
+
+    @property
+    def is_editable(self):
+        if self.browser.get_attribute("readonly", self.text):
+            return False
+        else:
+            return True
+
+    def __init__(self, parent, locator=None, logger=None):
+        super().__init__(parent, logger=logger)
+        if locator:
+            self.locator = locator
+        else:
+            self.locator = self.DEFAULT_LOCATOR
+
+
+class ClipboardCopy(BaseClipboardCopy, View):
+    ROOT = ParametrizedLocator("{@locator}")
+    DEFAULT_LOCATOR = ".//div[contains(@class,'pf-c-clipboard-copy')]"
+
+    text = TextInput(locator=BaseClipboardCopy.TEXT_LOC)
+    button = Button(locator=BaseClipboardCopy.BUTTON_LOC)
+
+    def read(self):
+        return self.text.value
+
+    def fill(self, value):
+        self.text.fill(value)
```

Comment:
It's good to return as TextInput returning a boolean for the fill method.

Reply:
I don't understand what you mean, can you show me an example please?

Reply:
[TextInput](https://github.com/RedHatQE/widgetastic.core/blob/master/src/widgetastic/widget/input.py#L57) widget fill method return boolean means if value is filled then `True` else `False`

so I think we should return that value...

```
return self.text.fill(value)
```

Reply:
Ah, yeah works, I also added an assert to the test.